universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -321,63 +321,128 @@ private ContainerTag blockEditPanel(
     }
 
-    String deleteQuestionAction =
-        controllers.admin.routes.AdminProgramBlockQuestionsController.destroy(program.id(), blockId)
-            .url();
-    ContainerTag questionDeleteForm =
-        form(csrfTag)
-            .withId("block-questions-form")
-            .withMethod(HttpVerbs.POST)
-            .withAction(deleteQuestionAction);
+    ContainerTag programQuestions = div();
     blockQuestions.forEach(
-        pqd -> questionDeleteForm.with(renderQuestion(pqd.getQuestionDefinition(), canDelete)));
+        pqd ->
+            programQuestions.with(
+                renderQuestion(
+                    csrfTag,
+                    program.id(),
+                    blockId,
+                    pqd.getQuestionDefinition(),
+                    canDelete,
+                    pqd.required())));
 
     return div()
         .withClasses(Styles.FLEX_AUTO, Styles.PY_6)
-        .with(blockInfoDisplay, buttons, questionDeleteForm);
+        .with(blockInfoDisplay, buttons, programQuestions);
   }
 
-  private ContainerTag renderQuestion(QuestionDefinition definition, boolean canRemove) {
+  private ContainerTag renderQuestion(
+      Tag csrfTag,
+      long programDefinitionId,
+      long blockDefinitionId,
+      QuestionDefinition questionDefinition,
+      boolean canRemove,
+      boolean isRequired) {
     ContainerTag ret =
         div()
             .withClasses(
-                Styles.RELATIVE,
+                ReferenceClasses.PROGRAM_QUESTION,
                 Styles.MX_4,
                 Styles.MY_2,
                 Styles.BORDER,
                 Styles.BORDER_GRAY_200,
                 Styles.PX_4,
                 Styles.PY_2,
                 Styles.FLEX,
-                Styles.ITEMS_START,
+                Styles.GAP_4,
+                Styles.ITEMS_CENTER,
                 canRemove ? "" : Styles.OPACITY_50,
                 StyleUtils.hover(Styles.TEXT_GRAY_800, Styles.BG_GRAY_100));
 
+    ContainerTag icon =
+        Icons.questionTypeSvg(questionDefinition.getQuestionType(), 24)
+            .withClasses(Styles.FLEX_SHRINK_0, Styles.H_12, Styles.W_6);
+    ContainerTag content =
+        div()
+            .withClass(Styles.FLEX_GROW)
+            .with(p(questionDefinition.getName()))
+            .with(p(questionDefinition.getDescription()).withClasses(Styles.MT_1, Styles.TEXT_SM));
+
+    ContainerTag requiredButton =
+        TagCreator.button()
+            .withClasses(
+                Styles.FLEX,
+                Styles.GAP_1,
+                Styles.ITEMS_CENTER,
+                isRequired ? Styles.TEXT_BLACK : Styles.TEXT_GRAY_400,
+                Styles.FONT_MEDIUM,
+                Styles.BG_TRANSPARENT,
+                Styles.ROUNDED_FULL,
+                StyleUtils.hover(Styles.BG_GRAY_400, Styles.TEXT_GRAY_300))
+            .withType("submit")
+            .with(p("required").withClasses("hover-group:text-white"))
```

Comment:
question (non-blocking): I think in the mocks this read "optional", does it matter?

Reply:
it doesn't. we can flip it on the front end once we do more styling. 

my two cents: it seems like `required` should be what is shown, if that is the special case. If questions are default optional, then the toggle should be to turn on required.